NEW: Astronomer Certified Versioning + Support Policy

paolaperaza opened the pull request:
In this PR, I'm addressing a long-standing need to publish our versioning and support policy for Astronomer Certified. Towards that effort, this PR does the following:

- Adds a new doc (applicable to Cloud + Enterprise) that details AC versioning + support schedule
- Removes 1.10.5 + 1.10.7 from our "Upgrade Airflow" doc and links out to this new doc

@jwitz I think my first pass here is all set. Do you wanna take a look here and let me know what you think? If everything looks good, we can pass these changes onto Enterprise docs + potentially pull out the AC section of the "Enterprise Compatibility" doc as well.

cc: @vikramkoka + @krisdock I'd love both of your reviews here as well. This PR is largely based on the Notion doc here: https://www.notion.so/astronomerio/Astronomer-Certified-End-of-Life-Policy-4b5acca2db0645a1b65f2fa866c281a1

Addresses: https://github.com/astronomer/docs/issues/371

krisdock commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+| [2.0.0](https://github.com/astronomer/ap-airflow/blob/master/2.0.0/CHANGELOG.md)     | FROM quay.io/astronomer/ap-airflow:2.0.0-buster-onbuild   | N/A                                                           |
+| [2.0.2](https://github.com/astronomer/ap-airflow/blob/master/2.0.2/CHANGELOG.md)     | FROM quay.io/astronomer/ap-airflow:2.0.2-buster-onbuild   | N/A                                                           |
+| [2.1.0](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md)     | FROM quay.io/astronomer/ap-airflow:2.1.0-buster-onbuild   | N/A                                                           |
+| [2.1.1](https://github.com/astronomer/ap-airflow/blob/master/2.1.1/CHANGELOG.md)     | FROM quay.io/astronomer/ap-airflow:2.1.1-buster-onbuild   | N/A                                                           |
```
2.1.3 was released recently: https://github.com/astronomer/ap-airflow/blob/master/2.1.3/CHANGELOG.md

krisdock commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+| 2.0.0      | Dec 17, 2020   | 18 Months      | June 17, 2022       |
+| 2.0.2      | April 26, 2021 | 18 Months      | Oct 26, 2022        |
+| 2.1.0      | May 21, 2021   | 18 Months      | Nov 21, 2022        |
+| 2.1.1      | July 2, 2021   | 18 Months      | Jan 2, 2023         |
```
same here, add 2.1.3

krisdock commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+- 2.1.0-1
+- 2.1.0-2
+- 2.1.0-3 etc.
```
this talks about 1.10.10, but lists hotfixes for 2.1.0

krisdock commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+All hotfix releases are listed in a [corresponding changelog](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md), which specifies the date the hotfix was released and all individual changes made to it. Bugs that are reported by the wider Airflow community are often backported by our team and made available prior to the subsequent open source release.
+
+For information on how to upgrade to latest hotfix release, read [Upgrade to an AC Patch Version](https://www.astronomer.io/docs/cloud/stable/customize-airflow/manage-airflow-versions#patch-versions-of-astronomer-certified).
```
upgrade to >the< latest

krisdock reviewed (commented):
left a few comments

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+- `quay.io/astronomer/ap-airflow:2.1.0-buster`
+- `quay.io/astronomer/ap-airflow:2.1.0-buster-onbuild`
+
+For the smoothest, out-of-the-box Airflow experience, we strongly recommend and default to `buster-onbuild` images in your project's `Dockerfile`. These images incorporate Docker ONBUILD commands to copy your Airflow project directory (i.e. your `packages.txt`, `requirements.txt`, `dags`, etc.) and scaffold it so you can more easily pass those files to the containers running each core Airflow com ...
```
```suggestion
For the smoothest, out-of-the-box Airflow experience, we strongly recommend and default to `buster-onbuild` images in your project's `Dockerfile`. These images incorporate Docker ONBUILD commands to copy your Airflow project directory and scaffold it so you can more easily pass those files to the containers running each core Airflow component.
```

jwitz replied:
Doesn't seem quite like the place to add this level of detail

jwitz replied:
Bumping this

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+For the smoothest, out-of-the-box Airflow experience, we strongly recommend and default to `buster-onbuild` images in your project's `Dockerfile`. These images incorporate Docker ONBUILD commands to copy your Airflow project directory (i.e. your `packages.txt`, `requirements.txt`, `dags`, etc.) and scaffold it so you can more easily pass those files to the containers running each core Airflow com ...
+
+For complex use cases that require that you customize the AC base image, read [Customize your Airflow Image on Astronomer](https://www.astronomer.io/docs/cloud/stable/develop/customize-image).
```
```suggestion
For complex use cases that require customizing the base AC image, read [Customize your Airflow Image on Astronomer](https://www.astronomer.io/docs/cloud/stable/develop/customize-image).
```

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+### Hotfix Versions
+
+In addition to supporting the latest versions of open source Airflow on Astronomer Certified (AC), our team regularly ships bug and security hotfixes to AC images as hotfix releases.
```
```suggestion
In addition to supporting the latest versions of open source Airflow on AC, Astronomer regularly ships bug and security fixes to AC images in hotfix releases.
```

Since H3 is a continuation of h2 and can't be linked out, we don't need to reintroduce the acronym here.

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+In addition to supporting the latest versions of open source Airflow on Astronomer Certified (AC), our team regularly ships bug and security hotfixes to AC images as hotfix releases.
+
+For example, Astronomer Certified 1.10.10 has been enhanced with 4 additional patches since its initial release:
```
```suggestion
The format for AC hotfix versions is `<base-version>-<hotfix-version>. For example, the second hotfix release for AC 2.1.0 was `2.1.0-2`.
```

jwitz replied:
If you take this suggestion, remove the bullets below.

jwitz replied:
Bumping

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+- 2.1.0-2
+- 2.1.0-3 etc.
+
+All hotfix releases are listed in a [corresponding changelog](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md), which specifies the date the hotfix was released and all individual changes made to it. Bugs that are reported by the wider Airflow community are often backported by our team and made available prior to the subsequent open source release.
```
```suggestion
All hotfix releases have a [corresponding changelog](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md) which specifies the date the hotfix was released and all individual changes made to it. Bugs that are reported by the wider Airflow community are often backported  and made available prior to the subsequent open source release.
```
Is it still "backported" if we get them before the official release?

jwitz replied:
Bumping for "backported" question

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+## Available Versions
+
+Below you'll find a matrix of all Astronomer Certified images supported on Astronomer Cloud. For our platform's full collection of Docker images, reference [Astronomer on Quay.io](https://quay.io/repository/astronomer/ap-airflow?tab=tags). For more information on Alpine and Debian as distinct system distributions, read [Migrate from Alpine to Debian](https://www.astronomer.io/docs/cloud/stable/cu ...
```
```suggestion
The following table contains all Astronomer Certified images supported on Astronomer Cloud. For our platform's full collection of Docker images, reference [Astronomer on Quay.io](https://quay.io/repository/astronomer/ap-airflow?tab=tags). For more information on Alpine and Debian as distinct system distributions, read [Migrate from Alpine to Debian](https://www.astronomer.io/docs/cloud/stable/customize-airflow/manage-airflow-versions#migrate-from-alpine-to-debian).
```

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+Astronomer is committed to supporting all patch versions of Astronomer Certified (AC) for a total of 18 months from the release date of the Docker image and Python wheel. Within the support window of each Airflow + Astronomer Certified version, the following is true:
+
+- Astronomer Certified Docker images and Python wheel for that version are available for first-time download via [Quay.io](http://quay.io), PyPi and [astronomer.io/downloads](https://support.astronomer.io).
```
```suggestion
- AC Docker images and Python wheel for a version are available for first-time download via [Quay.io](http://quay.io), PyPi and [astronomer.io/downloads](https://support.astronomer.io).
```
What does "first-time" mean here?

paolaperaza replied:
Addressed here - lmk what you think, agreed that "first time download" is a bit redundant

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+When a version of Airflow is no longer supported, the following is true:
+
+- The Astronomer Support team is not obligated to answer questions regarding an Airflow Deployment that is running a version of Airflow or Astronomer Certified that is no longer supported.
```
```suggestion
- The Astronomer Support team is not obligated to answer questions regarding an Airflow Deployment that is running an outdated version.
```

paolaperaza replied:
I think we want to be a bit strong-handed here. I wouldn't say a version is "outdated" - that's more of a subjective term? I think "no longer supported" is accurate but let me know if you have a better suggestion to be clearer. Otherwise considered, "that has reached its end of life date".

jwitz replied:
Agreed! In that case I would cut down of the length by saying "an unsupported version of Airflow or Astronomer Certified"

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+When a version of Airflow is no longer supported, the following is true:
+
+- The Astronomer Support team is not obligated to answer questions regarding an Airflow Deployment that is running a version of Airflow or Astronomer Certified that is no longer supported.
+- New Airflow Deployments cannot be created with a version of AC that is no longer supported. Deprecated versions will *not* render as an option in the Deployment creation process from the Astronomer UI, API or CLI.
```
```suggestion
- New Airflow Deployments cannot be created with a version of AC that is no longer supported. Deprecated versions will *not* render as an option in the Deployment creation process from the Astronomer UI, API, or CLI.
```

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+- Astronomer Certified Docker images and Python wheel for that version are available for first-time download via [Quay.io](http://quay.io), PyPi and [astronomer.io/downloads](https://support.astronomer.io).
+- Astronomer will regularly publish hotfixes for bug or security issues identified as high priority.
+- The Astronomer Support team will offer support for paying customers running a supported version of AC via the [Astronomer Support Portal](https://support.astronomer.io).
+- A user can create a new Airflow Deployment via the Astronomer UI, CLI, or API with any supported version of Astronomer Certified.
```
```suggestion
- A user can create a new Airflow Deployment via the Astronomer UI, CLI, or API with any supported version of AC.
```

jwitz commented on cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md:
```diff
@@ -0,0 +1,99 @@
+
+## Support Schedule
+
+Per Astronomer Certified's 18-month support lifecycle, the exact support timeline for each published version is defined below.
```
```suggestion
The following table contains the exact support timeline for each published version. These timelines are based on Astronomer Certified's 18-month support lifecycle.
```

jwitz reviewed (commented):
Mostly wording suggestions, but I also dropped a few questions throughout. Let me know what you think!

paolaperaza commented:
Thank you for the review, @krisdock ! @jwitz I think I addressed all of the comments above and we just need to bring this over to Enterprise docs. I'm a bit tight on time tomorrow but would love to get this out - any chance you could do that in the morning?

Appreciate you both 🙏

jwitz commented on cloud/stable/06_astronomer-certified/03_support-policy.md:
```diff
@@ -0,0 +1,102 @@
+- 2.1.0-2
+- 2.1.0-3 etc.
+
+All hotfix releases have a [corresponding changelog](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md), which specifies the date the hotfix was released and all individual changes made to it. Bugs that are reported by the wider Airflow community are often backported by our team and made available prior to the subsequent open source release.
```
```suggestion
All hotfix releases have a [corresponding changelog](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md) which specifies the date the hotfix was released and all individual changes made to it. Bugs that are reported by the wider Airflow community are often backported by our team and made available prior to the subsequent open source release.
```
No comma because everything after "which" is a dependent clause.

jwitz reviewed (approved):
I bumped a few comments that were unaddressed and responded to some other comments. Once the bumped comments are addressed this will be good to go @paolaperaza !
